src/Microsoft.Extensions.Localization/IApplicationNameProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Localization
+{
+    public interface IApplicationNameProvider
```

Comment:
This problem with this approach is that your `AddLocalization` methods no longer work out of the box. This removes the dependency on Hosting, but replaces it with an implicit dependency on something else. 

Why not add the application name to `LocalizationOptions` and do both of:
- define the behavior to something sane when application name isn't set
- teach MVC to set the application name for your when localization is in use

This doesn't require any new abstractions or plumbing, and has the advantage that localization can be used outside of MVC.